_posts/2021-04-14-accessing-the-relay-store.markdown

```diff
@@ -0,0 +1,260 @@
+
+Relay is a GraphQL client library maintained by Facebook engineers and enables rapid client-side data fetching in
+React applications. [Artsy's adoption of Relay][why-does-artsy-use-relay] coincided with our move toward using
+React Native for our mobile work around 2016. I joined Artsy as an engineer in November of 2020 (after
+[transitioning to engineering from a non-technical role at the
+company][how-losing-my-way-helped-me-find-my-way-back].) When I joined, I was about a year into React development
+and completely new to Relay.
+
+I work on the Partner Experience (PX) team at Artsy. We build and maintain software used by our gallery and auction
+house partners to sell artwork on Artsy. Although Relay is not particularly new to Artsy, it’s relatively new to
+our team’s main repository, Volt. (Volt is Artsy’s CMS used by gallery partners to manage their presences on the
+platform.) A topic for another blog post, but Volt’s structure is worth noting here. Volt is a collection of mini
+React apps injected into HAML views——our way of incrementally converting the codebase to our new stack.
+
+Relay is one of the many new pieces of technology I've learned on the job here at Artsy. Relay’s biggest advantage
+in my eyes is how it tightly couples the client view and API call (in our case, to the GraphQL layer of our stack,
+which we call Metaphysics.) In addition to performance and other benefits, having the data fetch right on top of
+the React component creates a pretty seamless developer experience.
+
+### Building an Artwork Checklist
+
+On PX, we recently launched a checklist feature aimed at empowering our gallery partners to be more self-sufficient
+and find the greatest success possible on Artsy. The checklist prompts galleries to add specific metadata to
+artworks that we know (because of our awesome data team) will make the work more likely to sell. The new feature
+gathers a list of five high-priority artworks (meaning they are published, for-sale, and by a top-selling artist)
+that are missing key pieces of metadata. Users are prompted to add the pieces of meatdata and also have the ability
+to click a button to “snooze” works, which removes them from the list for the day.
+
+<figure class="illustration">
+    <img src="https://user-images.githubusercontent.com/9466631/114630150-28886200-9c77-11eb-9f04-461101496ee0.png">
+</figure>
+
+The feature makes use of [Redis][about-redis], a key-value store used for in-memory cache, to store two lists:
+
+1. `includeIDs` to store the five artworks in the list, so users see a consistent list of artworks whenever they
+   log in and load the page
+2. `excludeIDs` or “snoozed” IDs which Redis will store for 24 hours and ensure the user does not see
+
+When a user presses the “snooze” button, the ID for the artwork is added to the snoozed list in Redis. The list
+`includeIDs` and the list of `excludeIDs` are passed down from Rails controllers to our HAML views and then passed
```

Comment:
There's a gap between `includeIDs` and the actual name in the example code: `checklistArtworkIDs`. (There's also a smaller gap between `excludeIDs` and `checklistExcludedArtworkIDs` but that one is easier to traverse due to `exclude` being in both.)

Consistency between the names in this sentence and the code sample will make it easier to understand. Even though it wouldn't compile against our schema, I'd lean toward as simple of names as possible to illustrate the point.